font/src/lib.rs

```diff
@@ -135,39 +140,48 @@ impl FontKey {
     }
 }
 
-#[derive(Debug, Copy, Clone, Eq)]
-pub struct GlyphKey {
-    pub c: char,
-    pub font_key: FontKey,
-    pub size: Size,
+/// Captures possible outcomes of shaping, if shaping succeeded it will return a `GlyphIndex`.
+/// If shaping failed or did not occur, `Fallback` will be returned.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum KeyType {
+    /// Shaping returned a valid index and we can render that as expected.
+    GlyphIndex(u32),
+    /// Shaping returned a missing glyph or shaping did not occur. If glyph is missing system will
+    /// attempt to load character glyph from a fallback font. If shaping did not occur this will
+    /// first try the configured font then fallback.
+    Fallback(char),
```

Comment:
I'd like a clarification here since I'm not entirely certain how things work and it seems like the comments/naming are not sufficiently explaining this at the moment.

If I understand things correctly, on non-harfbuzz systems we do never use the GlyphIndex and on harfbuzz supported systems we only use `Fallback` for font fallback, right?

That's what I remembered and the comments seem to support this, however I don't think the `Fallback` name sufficiently explains this. One might think that this is only used for font fallback, but that's not accurate. So I'd like to steer away from it.

Maybe just having this as `KeyType::Char` would be better? It's simple and accurately describes the type, without going into too much detail what it might be used for. I think that might be a bit better? But alternatives are certainly possible. Maybe `RawChar` if we want to steer away from using just `Char`, but since it's capitalized I don't think it should create an issue.

Very curious what you'd think about this, it just seems that `Fallback` currently isn't ideal.

Reply:
I think that's a good point, originally it was intended to specify that the variant immediately fell back to system font and didn't check the user font at all. This was also made while everything was still under the hb-ft flag so it was assumed things were harfbuzz. However based on what it looks like font ligatures will look like under windows (and probably mac) I agree it no longer makes sense.

TL;DR I'll change it to Char and linux font fetching code can be commented to explain the fallback behavior.